### PR TITLE
Add branch section to `README.md`

### LINKED ISSUE #1189: Update `README.md` with section about branches
Hornet has a small section in their `README.md` that explains the purpose of the different branches in the repo: https://github.com/gohornet/hornet

We maybe should add something similar.

### PR DESCRIPTION
**Pull request by grtlr**
Closes #1189.

## Type of change

- Documentation Fix

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,19 @@ We have chosen the Rust programming language for Bee because of its C/C++ like p
 
 :::
 
+## Development
+
+The Bee repository has different branches:
+
+|Branch|Description|
+|------|-----------|
+|`mainnet`|The latest release for the IOTA protocol|
```
Comment (thibault-martinez) [inline README.md]:
```suggestion
|`mainnet`|The latest release for the IOTA network|
```

Comment (grtlr) [reply README.md]: Yeah, we have the naming mixed up. Each branch actually contains many _networks_ (each with their respective `config.*.json`). Shimmer will also have a `mainnet`, `testnet`, and so on. I think `mainnet` is the wrong name here, it should probably be just `iota`.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,19 @@ We have chosen the Rust programming language for Bee because of its C/C++ like p
...
+|`mainnet-develop*`|Ongoing development for the IOTA protocol|
```
Comment (pvdrz) [inline README.md]: why the `*`?

Comment (grtlr) [reply README.md]: Because we have `mainnet-develop` and `mainnet-develop-0.4` right now.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,17 @@ We have chosen the Rust programming language for Bee because of its C/C++ like p
 
 :::
 
+## Development
+
+The Bee repository has different branches:
+
+|Branch|Description|
+|------|-----------|
+|`production`|The latest release for the IOTA networks (`mainnet`, `devnet`, `comnet`).|
```
Comment (thibault-martinez) [inline README.md]:
```suggestion
|`production`|The latest release for the IOTA networks.|
```
I wouldn't necessarily name them.

File: README.md [hunk 4]
```diff
@@ -40,6 +40,17 @@ We have chosen the Rust programming language for Bee because of its C/C++ like p
...
+|`develop`|The ongoing development for future releases of these networks. With every release, the `develop` branch will be merged into `production`.|
+|`staging`|The latest release for the Shimmer networks (`testnet`).|
+| other | Branches with codenames like `stardust` reflect current projects. Similar to `develop` these project branches will find their way into `staging` once they are ready.|
```
Comment (grtlr) [inline README.md]:
```suggestion
| other | Branches with codenames like `stardust` reflect current projects. Similar to `develop`, they will find their way into `staging` once they are ready.| 
```

File: README.md [hunk 5]
```diff
@@ -40,6 +40,17 @@ We have chosen the Rust programming language for Bee because of its C/C++ like p
...
 
+## Development
+
+The Bee repository has different branches:
+
+|Branch|Description|
+|------|-----------|
+|`production`|The latest release for the IOTA networks (`mainnet`, `devnet`, `comnet`).|
+|`develop`|The ongoing development for future releases of these networks. With every release, the `develop` branch will be merged into `production`.|
+|`staging`|The latest release for the Shimmer networks (`testnet`).|
```
Comment (thibault-martinez) [inline README.md]: Same, we may add/remove networks and we would always have to remember about that.

Comment (thibault-martinez) [reply README.md]: Sorry I wanted do to a suggestion to remove but it didn't work

Comment (grtlr) [reply README.md]:
```suggestion
|`staging`|The latest release for the Shimmer networks.|
```

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**